app/models/concerns/foreman/sti.rb

```diff
@@ -8,11 +8,11 @@ class << base
 
     module ClassMethods
       # ensures that the correct STI object is created when :type is passed.
-      def new(*attributes, &block)
-        if (h = attributes.first).is_a?(Hash) && (type = h.with_indifferent_access.delete(:type)) && !type.empty?
+      def new(attributes = nil, &block)
```

Comment:
It took me a while to figure out where it's defined, but `show-source ActiveRecord::Base.new` in pry helped. This is indeed the correct definition, since https://api.rubyonrails.org/classes/ActiveRecord/Inheritance/ClassMethods.html#method-i-new uses the same.